ext/mysql2/client.c

```diff
@@ -1613,6 +1641,21 @@ void init_mysql2_client() {
       INT2NUM(0));
 #endif
 
+#ifdef CLIENT_SESSION_TRACK
+  rb_define_method(cMysql2Client, "session_track", rb_mysql_client_session_track, 1);
```

Comment:
The pattern I'd like to follow is to always define this method on the object, and within the method, if the feature is unavailable, to return Qnil.